plugins/github/api.ml

```diff
@@ -94,35 +94,71 @@ module Status = struct
 end
 
 module CheckRunStatus = struct
-  type conclusion = [`Failure of string | `Success]
+  type conclusion = [`Failure of string | `Success | `Skipped of string]
   type state = [`Queued | `InProgress | `Completed of conclusion]
+  type action = { label:string; description:string; identifier:string }
   type t = {
       state: state;
-      description: string option;
+      summary: string option;
+      text: string option; 
+      actions: action list;
       url: Uri.t option
     }
 
-  let v ?description ?url state =
-    let description = Option.map (Astring.String.with_range ~len:140) description in (* Max GitHub allows *)
-    { state; description; url }
+  let action ~label ~description ~identifier = {label; description; identifier}
 
-  let state_to_string = function
+  let action_to_json {label; description; identifier} =
+    `Assoc [ ("label", `String label)
+           ; ("description", `String description)
+           ; ("identifier", `String identifier)]
+
+  let v ?text ?summary ?url ?(actions=[]) state =
+    (* A maximum of three actions are accepted by GitHub. *)
+    { state; summary; text; actions; url }
+
+  let state_to_string description summary actions = function
```

Comment:
This is badly named; it doesn't return a string!

Reply:
Reworked this.